src/lib/hooks/schedulerFactory.ts

```diff
@@ -0,0 +1,54 @@
+import { GroupChannel } from '@sendbird/chat/groupChannel';
+import { Logger } from '../SendbirdState';
+
+const TIMEOUT = 2000;
+
+export function schedulerFactory<T>({
+  logger,
+  timeout,
+  cb,
+}: {
+  logger: Logger;
+  timeout?: number;
+  cb: (item: T) => void;
+}) {
+  let queue: GroupChannel[] = [];
+  let interval: ReturnType<typeof setTimeout> | null = null;
+  const push = (channel: GroupChannel) => {
+    const channelPresent = queue.find((c) => c.url === channel.url);
+    if (!channelPresent) {
+      queue.push(channel);
+    } else {
+      logger.info('Channel: Mark as read already in queue', { channel });
+    }
+    // start the interval if it's not already running
+    if (interval) {
+      return;
+    }
+    const item = queue.shift();
+    cb(item as T);
```

Comment:
if there is no item on the queue, execuete immediately

Reply:
here's no condition to check if there's no item